pkg/clustermesh/types/types.go

```diff
@@ -52,3 +59,13 @@ func (c0 *CiliumClusterConfig) IsCompatible(c1 *CiliumClusterConfig) error {
 	}
 	return nil
 }
+
+// GetCapabilities returns the capabilities advertized through the given
+// CiliumClusterConfig, providing a default fallback in case it is nil.
+func (c *CiliumClusterConfig) GetCapabilities() CiliumClusterConfigCapabilities {
+	if c == nil {
```

Comment:
nit: relying on typed nils can introduce panics if someone decides to use to swap the *CiliumClusterConfig implementation out with an interface in the future.

Reply:
I added this method for convenience, because the retrieved CiliumClusterConfig is nil when not configured by the remote cluster. But I agree that it could lead to misleading behaviors. I'll remove it and move the logic to the caller.

Reply:
Dropped